core/src/main/scala/caliban/GraphQL.scala

```diff
@@ -117,19 +117,20 @@ trait GraphQL[-R] { self =>
                                       }
                   typeToValidate    = if (intro) introspectionRootType else rootType
                   schemaToExecute   = if (intro) introspectionRootSchema else schema
-                  validatedRequest <- VariablesCoercer.coerceVariables(request, doc, typeToValidate, skipValidation)
-
-                  validate          = (doc: Document) =>
-                                        Validator
-                                          .prepare(
-                                            doc,
-                                            typeToValidate,
-                                            schemaToExecute,
-                                            validatedRequest.operationName,
-                                            validatedRequest.variables.getOrElse(Map.empty),
-                                            skipValidation
-                                          )
-                  executionRequest <- wrap(validate)(validationWrappers, doc)
+                  validate          = (input: ValidationWrapperInput) =>
+                                        VariablesCoercer
+                                          .coerceVariables(request, input.document, typeToValidate, input.skipValidation)
+                                          .flatMap { validatedRequest =>
+                                            Validator.prepare(
+                                              input.document,
+                                              typeToValidate,
+                                              schemaToExecute,
+                                              validatedRequest.operationName,
+                                              validatedRequest.variables.getOrElse(Map.empty),
+                                              input.skipValidation
+                                            )
+                                          }
```

Comment:
Shall we just skip the validation wrappers if `skipValidation` is true? 🤔

Reply:
I think that's a good idea, although what worries me is whether there are any users that have validation wrappers (either custom or provided) and specify `skipValidation=true`. I'm happy to make the change if we think that won't be an issue.

A potential safe(r) way to do it (albeit somewhat messy) would be to add another field to the [ValidationWrapperInput](https://github.com/ghostdogpr/caliban/pull/1557/files/3883d333ced66bb8bcbcf290f6e1b9051de0a536#diff-8ffb6cba0c54bbf92fc33727497416acdd30f5a0c20c20109ccf5a79908c3512R153) (e.g., `skipValidationWrappers: Boolean`).

Reply:
Actually, now that I think about it a bit more, I don't think we can skip validation wrappers altogether. Ones like the `CostEstimation.queryCost` and `ApolloTracing.apolloTracing` might be a bit too messy to skip

Reply:
@ghostdogpr I have some local changes to skip validation wrappers (ones that make sense) when `skipValidation=true`. Do you prefer if I add them to this PR or to a followup one so that we can discuss them separately?

Reply:
Maybe do all here if it's a breaking change on the same thing

Reply:
Done!